src/pages/transactions/components/SearchArea.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { InputField } from '@invoiceninja/forms';
+import { Dispatch, SetStateAction, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { BiPlusCircle } from 'react-icons/bi';
+import { MdFilterAlt } from 'react-icons/md';
+import { CreateExpenseCategoryModal } from './CreateExpenseCategoryModal';
+import { CreateVendorModal } from './CreateVendorModal';
+import { FilterModal } from './FilterModal';
+import { SearchInput } from './ListBox';
+
+interface Props {
+  searchParams: SearchInput;
+  setSearchParams: Dispatch<SetStateAction<SearchInput>>;
+  setSelectedIds: Dispatch<SetStateAction<string[] | undefined>>;
+  setIsFilterModalOpened: Dispatch<SetStateAction<boolean>>;
+  isInvoicesDataKey: boolean;
+  isFilterModalOpened: boolean;
+  dataKey: string;
+}
+
+export function SearchArea(props: Props) {
+  const [t] = useTranslation();
+
+  const [
+    isCreateExpenseCategoryModalOpen,
+    setIsCreateExpenseCategoryModalOpen,
+  ] = useState<boolean>(false);
+
+  const [isCreateVendorModalOpen, setIsCreateVendorModalOpen] =
+    useState<boolean>(false);
+
+  const handleChangeSearchParams = (
+    property: keyof SearchInput,
+    value: SearchInput[keyof SearchInput]
+  ) => {
+    props.setSearchParams((prevState) => ({
+      ...prevState,
+      [property]: value,
+    }));
+  };
+
+  return (
+    <>
+      <CreateExpenseCategoryModal
```

Comment:
This should probably be encapsulated within expense category selector.

Reply:
done